### PR TITLE
perf: [ANDROAPP-2572] Improve dataset table performance

### PR DESCRIPTION
**Pull request by Balcan**
[jira issue](https://jira.dhis2.org/browse/ANDROAPP-2572) -> Performance
[jira issue](https://jira.dhis2.org/browse/ANDROAPP-2716) -> Width selector

### CODE REVIEW

File: app/src/main/res/layout/item_progress.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            style="@style/TextPrimary"
+            android:layout_width="0dp"
+            android:layout_height="0dp"
+            android:gravity="center_vertical"
+            android:paddingStart="5dp"
+            android:text="@string/loading"
+            app:layout_constraintBottom_toBottomOf="@id/progress"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toEndOf="@id/progress"
+            app:layout_constraintTop_toTopOf="@id/progress" />-->
```
Comment (JaimeToca) [inline app/src/main/res/layout/item_progress.xml]: can we remove it?

File: app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetSection/DataSetSectionFragment.java [hunk 2]
```diff
@@ -95,77 +92,61 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
...
+        }
+    }
 
-            if (position != -1 && currentTablePosition.getValue() != position)
-                currentTablePosition.setValue(position);
-        });
-        currentTablePosition.setValue(0);
+    @Override
+    public void finishTableLoading() {
+//        binding.programProgress.setVisibility(View.GONE);
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetSection/DataSetSectionFragment.java]: is this GONE needed?

File: app/src/main/res/layout/fragment_dataset_section.xml [hunk 3]
```diff
@@ -73,15 +63,15 @@
...
+        <!--<ProgressBar
             android:id="@+id/program_progress"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             app:layout_constraintBottom_toTopOf="@+id/action_layout"
             app:layout_constraintEnd_toEndOf="parent"
             app:layout_constraintStart_toStartOf="parent"
             app:layout_constraintTop_toBottomOf="@+id/headerContainer"
-            app:progressColor="@{@color/colorPrimary}" />
+            app:progressColor="@{@color/colorPrimary}" />-->
```
Comment (JaimeToca) [inline app/src/main/res/layout/fragment_dataset_section.xml]: why no progress bar :(

File: app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetSection/DataSetSectionFragment.java (hunk 2 above)
Comment (MarinaCape) [inline app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetSection/DataSetSectionFragment.java]: This comment could be removed if it is not using the code line

File: app/src/main/res/layout/fragment_dataset_section.xml [hunk 4]
```diff
@@ -73,15 +63,15 @@
 
         </androidx.constraintlayout.widget.ConstraintLayout>
 
-        <ProgressBar
+        <!--<ProgressBar
```
Comment (MarinaCape) [inline app/src/main/res/layout/fragment_dataset_section.xml]: This commented code could be removed

File: app/src/main/res/layout/fragment_dataset_section.xml (hunk 3 above)
Comment (Balcan) [reply app/src/main/res/layout/fragment_dataset_section.xml]: It is now inside the recycler view :)

File: app/src/main/res/layout/item_progress.xml (hunk 1 above)
Comment (Balcan) [reply app/src/main/res/layout/item_progress.xml]: yep

File: app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetSection/DataSetSectionFragment.java (hunk 2 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetSection/DataSetSectionFragment.java]: not anymore. I will remove this

**Pull request closed without merging**